src/connect4/core.rs

```diff
@@ -257,64 +278,103 @@ impl Board {
     fn get_run_in_direction(&self, start: GridPosition, dir: GridPosition, team: i32) -> i32 {
         let mut dir_active = true;
         let mut rev_active = true;
-        let mut space_used = false;
+        let mut dir_spaces_used = 0;
+        let mut rev_space_used = false;
         let mut run_len = 1i32; //Start with dropped token
+        let mut potential_len = 1; //Assume potential length starts at 1 for dropped token
         let mut i = 1; //Start one beyond dropped token
         while run_len <= 4 && (dir_active || rev_active) {
             dir_active = dir_active && self.on_board(GridPosition::new(start.x+i*dir.x, start.y+i*dir.y));
             rev_active = rev_active && self.on_board(GridPosition::new(start.x-i*dir.x, start.y-i*dir.y));
             //Do reverse case first for edge case of AASA_A is treated as a run of 4 and not 3 with a space
             if rev_active {
-                if self.get_cell_team(GridPosition::new(start.x-i*dir.x, start.y-i*dir.y)) == team {
-                    run_len += 1;
-                } else {
+                let val = self.get_cell_team(GridPosition::new(start.x-i*dir.x, start.y-i*dir.y));
+                //If token not for team in cell, end of search in rev direction
+                if val != 0 && val != team {
                     rev_active = false;
+                //If no spaces used, either add to run_len and/or potential_len depending on if cell is empty or matches team
+                } else if !rev_space_used {
+                    if val == team {
+                        run_len += 1;
+                    } else {
+                        rev_space_used = true;
+                    }
+                    potential_len += 1;
+                //If space in rev direction found, just add to potential run len to track that
+                } else {
+                    potential_len += 1;
```

Comment:
I forgot - I also reorganized this helper method. Slightly simpler code and logic was easier to test.